test/TestPercentageMath.sol

```diff
@@ -159,15 +165,28 @@ contract TestPercentageMath is Test {
         uint16 percentage
     ) public {
         vm.assume(percentage <= PERCENTAGE_FACTOR);
-        if (percentage > 0) vm.assume(y <= MAX_UINT256_MINUS_HALF_PERCENTAGE / percentage);
-        if (percentage < PERCENTAGE_FACTOR)
-            vm.assume(x <= MAX_UINT256_MINUS_HALF_PERCENTAGE / (PERCENTAGE_FACTOR - percentage));
-
-        assertApproxEqAbs(
-            PercentageMath.weightedAvg(x, y, percentage),
-            ((x * (PERCENTAGE_FACTOR - percentage)) + (y * percentage)) / PERCENTAGE_FACTOR,
-            1
+        // prettier-ignore
+        vm.assume(
+            (percentage == 0 || y <= MAX_UINT256_MINUS_HALF_PERCENTAGE / percentage) &&
+            (PERCENTAGE_FACTOR - percentage == 0 || x <= MAX_UINT256_MINUS_HALF_PERCENTAGE / (PERCENTAGE_FACTOR - percentage))
+        );
+
+        assertEq(PercentageMath.weightedAvg(x, y, percentage), mathRef.weightedAvg(x, y, percentage));
+    }
+
+    function testFailWeightedAvgOverflow(
+        uint256 x,
+        uint256 y,
+        uint256 percentage
+    ) public {
+        vm.assume(percentage <= PERCENTAGE_FACTOR);
+        // prettier-ignore
+        vm.assume(
+            (percentage > 0 && y > MAX_UINT256_MINUS_HALF_PERCENTAGE / percentage) ||
+            (PERCENTAGE_FACTOR - percentage > 0 && x > (MAX_UINT256_MINUS_HALF_PERCENTAGE / (PERCENTAGE_FACTOR - percentage)))
         );
+
+        PercentageMath.weightedAvg(x, y, percentage);
     }
```

Comment:
this test should not pass right?

Reply:
hmm, we assume x and such as it should fail
and we expect it to fail 
it seems ok

Reply:
Naming `testFail` is a way to expect the test to revert??

Reply:
I faced some issues with foundry and the `vm.expectRevert` so perhaps @MathisGD can change it, just saying.

Reply:
Ok I can change it if you prefer. I don't mind